corehq/apps/users/views/mobile/users.py

```diff
@@ -679,6 +685,226 @@ def post(self, request, *args, **kwargs):
+        return {
+            'response': {
+                'itemList': map(self._format_user, users_data.hits),
+                'total': users_data.total,
+                'page': page,
+                'query': query,
+            },
+            'success': True,
+        }
+
+    @allow_remote_invocation
+    def modify_user_status(self, in_data):
+        if not self.can_add_extra_users:
+            return {
+                'error': _("No Permission."),
+            }
+        try:
+            user_id = in_data['user_id']
+        except KeyError:
+            return {
+                'error': _("Please provide a user_id."),
+            }
+        try:
+            is_active = in_data['is_active']
+        except KeyError:
+            return {
+                'error': _("Please provide an is_active status."),
+            }
+        user = CommCareUser.get_by_user_id(user_id, self.domain)
+        user.is_active = is_active
+        user.save()
+        return {
+            'success': True,
+        }
+
+    @allow_remote_invocation
+    def check_username(self, in_data):
+        try:
+            username = in_data['username'].strip()
+        except KeyError:
+            return HttpResponseBadRequest('You must specify a username')
+        if username == 'admin' or username == 'demo_user':
+            return {'error': _('Username {} is reserved.'.format(username))}
+        if '@' in username:
+            return {
+                'error': _('Username {} cannot contain "@".'.format(username))
+            }
+        if ' ' in username:
+            return {
+                'error': _('Username {} cannot contain '
+                           'spaces.'.format(username))
+            }
+        full_username = format_username(username, self.domain)
+        if CommCareUser.get_by_username(full_username, strict=True):
+            result = {'error': _('Username {} is already taken'.format(username))}
+        else:
+            result = {'success': _('Username {} is available'.format(username))}
+        return result
+
+    @allow_remote_invocation
+    def create_mobile_worker(self, in_data):
+        if not self.can_add_extra_users:
+            return {
+                'error': _("No Permission."),
+            }
+        try:
+            user_data = in_data['mobileWorker']
+        except KeyError:
+            return {
+                'error': _("Please provide mobile worker data."),
+            }
+
+        try:
+            form_data = {}
+            for k, v in user_data.get('customFields', {}).items():
+                form_data["{}-{}".format(CUSTOM_DATA_FIELD_PREFIX, k)] = v
+            for f in ['username', 'password', 'first_name', 'last_name']:
+                form_data[f] = user_data[f]
+            form_data['domain'] = self.domain
+            self.request.POST = form_data
+        except Exception as e:
+            return {
+                'error': _("Check your request: %s" % e)
+            }
```

Comment:
It feels weird to mutate `self.request` in the view.  Can this parsing be handled in the form validation?  Then if `not form.is_valid()`, you'll have access to the form errors.